stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -329,13 +332,15 @@ public void setMessageCompression(boolean enable) {
     }
   }
 
-  private static class StreamObserverToCallListenerAdapter<ReqT, RespT>
+  // Non private to avoid synthetic class
+  static final class StreamObserverToCallListenerAdapter<ReqT, RespT>
```

Comment:
This is only used in static methods of the outer class. I thought outer class have direct access to its private inner classes, thus only the constructor needed to be non-private. Isn't it the case?